flytekit/core/type_engine.py

```diff
@@ -1023,13 +1023,30 @@ def get_transformer(cls, python_type: Type) -> TypeTransformer[T]:
             # Special case: prevent that for a type `FooEnum(str, Enum)`, the str transformer is used.
             return cls._ENUM_TRANSFORMER
 
+        from flytekit.types.iterator.json_iterator import JSONIterator
+
         for base_type in cls._REGISTRY.keys():
             if base_type is None:
                 continue  # None is actually one of the keys, but isinstance/issubclass doesn't work on it
             try:
-                if isinstance(python_type, base_type) or (
-                    inspect.isclass(python_type) and issubclass(python_type, base_type)
+                origin_type: typing.Any | None = base_type
```

Comment:
Could we use Optional here instead? IIRC, python <3.10 doesn't support `|` 
```suggestion
                origin_type: Optional[typing.Any] = base_type
```